gammapy/makers/background/tests/test_phase.py

```diff
@@ -95,3 +99,127 @@ def test_check_phase_intervals(pars):
+                ),
+                np.reshape(np.array([163, 46, 15, 1, 0, 0]), (6, 1, 1)),
+            ],
+        },
+        {
+            "p_in": ["5029747", "not-point-like"],
+            "p_out": [
+                np.reshape(
+                    np.array(
+                        [
+                            23,
+                            18,
+                            2,
+                            2,
+                            0,
+                            0,
+                        ]
+                    ),
+                    (6, 1, 1),
+                ),
+                np.reshape(
+                    np.array(
+                        [
+                            51,
+                            35,
+                            15,
+                            1,
+                            0,
+                            0,
+                        ]
+                    ),
+                    (6, 1, 1),
+                ),
+            ],
+        },
+        {
+            "p_in": ["111630", "not-point-like"],
+            "p_out": [
+                np.reshape(
+                    np.array(
+                        [
+                            15,
+                            5,
+                            5,
+                            8,
+                            2,
+                            1,
+                        ]
+                    ),
+                    (6, 1, 1),
+                ),
+                np.reshape(
+                    np.array(
+                        [
+                            24,
+                            13,
+                            15,
+                            12,
+                            2,
+                            0,
+                        ]
+                    ),
+                    (6, 1, 1),
+                ),
+            ],
+        },
+    ],
+)
+@requires_data()
+def test_make_counts(observations, phase_bkg_maker, pars):
+
+    maker = SpectrumDatasetMaker(
+        containment_correction=False, selection=["counts", "exposure", "edisp"]
+    )
+
+    e_reco = MapAxis.from_energy_bounds(0.05, 100, nbin=6, unit="TeV", name="energy")
+    e_true = MapAxis.from_energy_bounds(
+        0.01, 300, nbin=15, unit="TeV", name="energy_true"
+    )
+
+    if pars["p_in"][0] == "111630":
+        pos = SkyCoord(128.83606354, -45.17643181, frame="icrs", unit="deg")
+        obs = observations[pars["p_in"][0]]
+    elif pars["p_in"][0] == "5029747":
+        pos = SkyCoord(083.6331144560900, +22.0144871383400, frame="icrs", unit="deg")
+        obs = observations[pars["p_in"][0]]
+        table = obs.events.table
+        table["PHASE"] = np.linspace(0, 1, len(table["TIME"]))
+        obs._events = EventList(table)
+
+    if pars["p_in"][1] == "point-like":
+        on_region = PointSkyRegion(pos)
+    elif pars["p_in"][1] == "not-point-like":
+        on_region = SphericalCircleSkyRegion(pos, radius=0.1 * u.deg)
```

Comment:
I imagine you can have the region directly in `p_in` instead of the `point-like` string

Reply:
Let me test this !

Reply:
I don't think I can do better that this for the geom, but tell me !